### PR TITLE
feat: Add checkbox to filter new challenges

### PR DESCRIPTION
**Pull request by sadakchap**
# Title and Issue number 

Title : [Challenges] Add checkbox to filter new challenges

Issue No. : https://github.com/sadanandpai/frontend-mini-challenges/issues/400

Close #<issue_no>

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] I have updated the Index.html file for my contribution

### CODE REVIEW

**Comment by sadakchap** [comment]
Something like this? @sadanandpai 

https://github.com/sadanandpai/frontend-mini-challenges/assets/44117648/acbf86d3-32a8-4a34-8f7a-8a83129627eb

File: apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx [hunk 1]
```diff
@@ -17,19 +17,24 @@ interface Props {
...
+    } else if (newChallenges) {
+      filteredChallenges = filterChallengeByKey([...challenges.values()], "isNew", true);
     } else {
-      setSortedChallenges(() => getSortedChallengesByDifficulty(challenges));
+      filteredChallenges = getSortedChallengesByDifficulty(challenges);
     }
-  }, [challenges, searchInput]);
+
+    setSortedChallenges(filteredChallenges);
+  }, [challenges, searchInput, newChallenges]);
```
Comment (sadanandpai) [inline apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:37]: doing it in multi-level, will that be efficient & cleaner?

filteredChallenges = filterByTitle(challenges, text)
newChallenges = filterByNew(filteredChallenges, isNew)

**Review by sadanandpai — APPROVED** [review]

**Comment by sadanandpai** [comment]
@sadakchap conflicts

**Comment by sadanandpai** [comment]
@sadakchap let me know if you help here to get unblocked

File: apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx (hunk 1 above)
Comment (sadakchap) [reply apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:37]: Hey @sadanandpai , not sure I follow you. Could you please elaborate the point here again?

**Comment by sadanandpai** [comment]
the PR is outdated. Closing it. Please raise a new PR if any plans to work on this. @sadakchap

**Comment by sadakchap** [comment]
> the PR is outdated. Closing it. Please raise a new PR if any plans to work on this. @sadakchap

Sure, I will open another pr for this. Could you please take a look at this comment? @sadanandpai

File: apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx (hunk 1 above)
Comment (sadanandpai) [reply apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx:37]: in steps I meant
`filteredChallenges`
`newChallenges`

if that helps to write cleaner code

**Pull request closed without merging**